app/src/main/java/org/openobservatory/ooniprobe/activity/MeasurementDetailActivity.java

```diff
@@ -225,6 +226,13 @@ public boolean onOptionsItemSelected(MenuItem item) {
                     json = new GsonBuilder().setPrettyPrinting().disableHtmlEscaping().create().toJson(new JsonParser().parse(json));
                     startActivity(TextActivity.newIntent(this, json));
                 } catch (Exception e) {
+                    if (OrchestraTask.getNetworkType(this).equals(OrchestraTask.NO_INTERNET)) {
+                        new MessageDialogFragment.Builder()
+                                .withTitle(getString(R.string.Modal_Error))
+                                .withMessage(getString(R.string.Modal_Error_RawDataNoInternet))
+                                .build().show(getSupportFragmentManager(), null);
+                        return true;
```

Comment:
I like this pattern you're using here more than the iOS one, wrt structured programming.

Reply:
It's the same pattern to launch alert I found around the app

Reply:
Yeah, I like this because it returns early.